web-ui/src/components/member-directory/MemberModal.jsx

```diff
@@ -59,31 +61,118 @@ const MemberModal = ({ member, open, onSave, onClose }) => {
     });
   };
 
-  const validateRequiredInputs = useCallback(() => {
-      return (editedMember.firstName?.length > 0 && editedMember.lastName?.length > 0
-         && editedMember.workEmail?.length > 0 && editedMember.title?.length > 0 && editedMember.location?.length >0 && editedMember.employeeId?.length > 0 && startDate )
-  }, [editedMember, startDate])
+  const validateInputs = useCallback(() => {
+    let regName = /^[a-zA-Z]+$/;
+    if (
+      !regName.test(
+        editedMember.firstName ||
+          !regName.test(editedMember.lastName) ||
+          !regName.test(editedMember.middleName)
+      )
+    ) {
+      dispatch({
+        type: UPDATE_TOAST,
+        payload: {
+          severity: "error",
+          toast:
+            "Please enter a valid name without numbers or special characters",
+        },
+      });
+      return false;
+    }
+    let regEmail = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!regEmail.test(editedMember.workEmail)) {
+      dispatch({
+        type: UPDATE_TOAST,
+        payload: {
+          severity: "error",
+          toast: "Please enter a valid email",
+        },
+      });
+      return false;
+    }
+    let todayYear = new Date().getFullYear();
+    if (
+      todayYear - editedMember.birthDay?.getFullYear() < 14 ||
+      todayYear - editedMember.birthDay?.getFullYear() > 125
+    ) {
+      dispatch({
+        type: UPDATE_TOAST,
+        payload: {
+          severity: "error",
+          toast: "Please enter a valid birthday",
+        },
+      });
+      return false;
+    }
+    if (
+      todayYear - editedMember.startDate?.getFullYear() > 80 ||
+      editedMember.startDate.getTime() > new Date().getTime()
+    ) {
+      dispatch({
+        type: UPDATE_TOAST,
+        payload: {
+          severity: "error",
+          toast: "Please enter a valid start date",
+        },
+      });
+      return false;
+    }
+    if (
+      editedMember.startDate?.getTime() >
+      editedMember.terminationDate?.getTime()
+    ) {
+      dispatch({
+        type: UPDATE_TOAST,
+        payload: {
+          severity: "error",
+          toast: "Please enter a start date before the termination date",
+        },
+      });
+      return false;
+    }
+    return true;
+  }, [editedMember, dispatch]);
+
+  const validateRequiredInputsPresent = useCallback(() => {
+    return (
+      editedMember.firstName?.length > 0 &&
+      editedMember.lastName?.length > 0 &&
+      editedMember.workEmail?.length > 0 &&
+      editedMember.title?.length > 0 &&
+      editedMember.location?.length > 0 &&
+      editedMember.employeeId?.length > 0 &&
```

Comment:
isn't birthdate a required field?

Reply:
The birthdate isn't marked as a required field anywhere, so I left it off the check.